crates/primitives/src/result.rs

```diff
@@ -217,6 +218,68 @@ pub enum InvalidTransaction {
     DepositSystemTxPostRegolith,
 }
 
+impl error::Error for InvalidTransaction {}
+
+impl fmt::Display for InvalidTransaction {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            InvalidTransaction::PriorityFeeGreaterThanMaxFee => {
+                write!(f, "Priority fee is greater than max fee")
+            }
+            InvalidTransaction::GasPriceLessThanBasefee => {
+                write!(f, "Gas price is less than basefee")
+            }
+            InvalidTransaction::CallerGasLimitMoreThanBlock => {
+                write!(f, "Caller gas limit exceeds the block gas limit")
+            }
+            InvalidTransaction::CallGasCostMoreThanGasLimit => {
+                write!(f, "Call gas cost exceeds the gas limit")
+            }
+            InvalidTransaction::RejectCallerWithCode => {
+                write!(f, "Reject transactions from senders with deployed code")
+            }
+            InvalidTransaction::LackOfFundForMaxFee { .. } => {
+                write!(f, "Lack of funds for max fee",)
+            }
+            InvalidTransaction::OverflowPaymentInTransaction => {
+                write!(f, "Overflow payment in transaction")
+            }
+            InvalidTransaction::NonceOverflowInTransaction => {
+                write!(f, "Nonce overflow in transaction")
+            }
+            InvalidTransaction::NonceTooHigh { .. } => write!(f, "Nonce too high"),
+            InvalidTransaction::NonceTooLow { .. } => write!(f, "Nonce too low"),
```

Comment:
Would be good to include values from NonceTooLow inside `write!`.

you can just append it to macro:
https://doc.rust-lang.org/beta/core/macro.write.html

Reply:
Ok added - wanted to avoid excess verbosity but I think this is very acceptable